Stack trace on startup: No up-and-running site-local (private) addresses found

Hi,

I'm trying to deploy on a version 1.5.1 cluster, and I'm getting the following stacks from the clients and masters:

```[2017-04-25T13:12:38,749][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [] uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupException: java.lang.IllegalArgumentException: No up-and-running site-local (private) addresses found, got [name:lo (lo), name:eth0 (eth0)]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:125) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:112) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.cli.SettingCommand.execute(SettingCommand.java:54) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:96) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.cli.Command.main(Command.java:62) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:89) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:82) ~[elasticsearch-5.1.1.jar:5.1.1]
Caused by: java.lang.IllegalArgumentException: No up-and-running site-local (private) addresses found, got [name:lo (lo), name:eth0 (eth0)]
	at org.elasticsearch.common.network.NetworkUtils.getSiteLocalAddresses(NetworkUtils.java:187) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.common.network.NetworkService.resolveInternal(NetworkService.java:246) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.common.network.NetworkService.resolveInetAddresses(NetworkService.java:220) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.common.network.NetworkService.resolveBindHostAddresses(NetworkService.java:130) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.transport.TcpTransport.bindServer(TcpTransport.java:585) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.transport.netty4.Netty4Transport.doStart(Netty4Transport.java:182) ~[?:?]
	at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:69) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.transport.TransportService.doStart(TransportService.java:188) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:69) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.node.Node.start(Node.java:577) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.bootstrap.Bootstrap.start(Bootstrap.java:234) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:308) ~[elasticsearch-5.1.1.jar:5.1.1]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:121) ~[elasticsearch-5.1.1.jar:5.1.1]
	... 6 more
```

This seems to be a not-uncommon exception seen when running ELS in K8S, but the fix isn't clear in the context of the operator.

--- a/example/example-es-cluster.json
+++ b/example/example-es-cluster.json
@@ -8,6 +8,7 @@
     "client-node-replicas": 3,
     "master-node-replicas": 2,
     "data-node-replicas": 3,
+    "network-host": "_local_",
     "zones": ["us-east-1c", "us-east-1d", "us-east-1e"],
     "data-volume-size": "10Gi",
     "java-options": "-Xms256m -Xmx256m",

--- a/pkg/k8sutil/k8sutil.go
+++ b/pkg/k8sutil/k8sutil.go
@@ -545,7 +545,7 @@ func (k *K8sutil) DeleteStatefulSet(clusterName string) error {
 
 // CreateClientMasterDeployment creates the client or master deployment
 func (k *K8sutil) CreateClientMasterDeployment(deploymentType, baseImage string, replicas *int32, javaOptions string,
-	resources myspec.Resources, imagePullSecrets []myspec.ImagePullSecrets, clusterName, statsdEndpoint string) error {
+	resources myspec.Resources, imagePullSecrets []myspec.ImagePullSecrets, clusterName, statsdEndpoint string, networkHost string) error {
 
 	component := fmt.Sprintf("elasticsearch-%s", clusterName)
 	discoveryServiceNameCluster := fmt.Sprintf("%s-%s", discoveryServiceName, clusterName)
@@ -649,6 +649,10 @@ func (k *K8sutil) CreateClientMasterDeployment(deploymentType, baseImage string,
 										Name:  "DISCOVERY_SERVICE",
 										Value: discoveryServiceNameCluster,
 									},
+									v1.EnvVar{
+										Name:  "NETWORK_HOST",
+										Value: networkHost,
+									},
 								},
 								Ports: []v1.ContainerPort{
 									v1.ContainerPort{
@@ -746,7 +750,7 @@ func TemplateImagePullSecrets(ips []myspec.ImagePullSecrets) []v1.LocalObjectRef
 
 // CreateDataNodeDeployment creates the data node deployment
 func (k *K8sutil) CreateDataNodeDeployment(replicas *int32, baseImage, storageClass string, dataDiskSize string, resources myspec.Resources,
-	imagePullSecrets []myspec.ImagePullSecrets, clusterName, statsdEndpoint string) error {
+	imagePullSecrets []myspec.ImagePullSecrets, clusterName, statsdEndpoint string, networkHost string) error {
 
 	fullDataDeploymentName := fmt.Sprintf("%s-%s", dataDeploymentName, clusterName)
 	component := fmt.Sprintf("elasticsearch-%s", clusterName)
@@ -836,6 +840,10 @@ func (k *K8sutil) CreateDataNodeDeployment(replicas *int32, baseImage, storageCl
 									v1.EnvVar{
 										Name:  "DISCOVERY_SERVICE",
 										Value: discoveryServiceNameCluster,
+									},
+									v1.EnvVar{
+										Name:  "NETWORK_HOST",
+										Value: networkHost,
 									},
 								},
 								Ports: []v1.ContainerPort{

--- a/pkg/processor/processor.go
+++ b/pkg/processor/processor.go
@@ -118,6 +118,7 @@ func (p *Processor) refreshClusters() error {
 				DataDiskSize:       cluster.Spec.DataDiskSize,
 				ElasticSearchImage: cluster.Spec.ElasticSearchImage,
 				JavaOptions:        cluster.Spec.JavaOptions,
+				NetworkHost:        cluster.Spec.NetworkHost,
 				Snapshot: myspec.Snapshot{
 					SchedulerEnabled: cluster.Spec.Snapshot.SchedulerEnabled,
 					BucketName:       cluster.Spec.Snapshot.BucketName,
@@ -183,8 +184,8 @@ func (p *Processor) processElasticSearchCluster(c *myspec.ElasticsearchCluster) 
 	p.k8sclient.CreateDataService(c.Metadata.Name)
 	p.k8sclient.CreateClientService(c.Metadata.Name, c.Spec.NodePort)
 
-	p.k8sclient.CreateClientMasterDeployment("client", baseImage, &c.Spec.ClientNodeReplicas, c.Spec.JavaOptions, c.Spec.Resources, c.Spec.ImagePullSecrets, c.Metadata.Name, c.Spec.Instrumentation.StatsdHost)
-	p.k8sclient.CreateClientMasterDeployment("master", baseImage, &c.Spec.MasterNodeReplicas, c.Spec.JavaOptions, c.Spec.Resources, c.Spec.ImagePullSecrets, c.Metadata.Name, c.Spec.Instrumentation.StatsdHost)
+	p.k8sclient.CreateClientMasterDeployment("client", baseImage, &c.Spec.ClientNodeReplicas, c.Spec.JavaOptions, c.Spec.Resources, c.Spec.ImagePullSecrets, c.Metadata.Name, c.Spec.Instrumentation.StatsdHost, c.Spec.NetworkHost)
+	p.k8sclient.CreateClientMasterDeployment("master", baseImage, &c.Spec.MasterNodeReplicas, c.Spec.JavaOptions, c.Spec.Resources, c.Spec.ImagePullSecrets, c.Metadata.Name, c.Spec.Instrumentation.StatsdHost, c.Spec.NetworkHost)
 
 	zoneCount := 0
 	if len(c.Spec.Zones) != 0 {
@@ -198,13 +199,13 @@ func (p *Processor) processElasticSearchCluster(c *myspec.ElasticsearchCluster) 
 		zoneDistribution := p.calculateZoneDistribution(c.Spec.DataNodeReplicas, zoneCount)
 
 		for index, count := range zoneDistribution {
-			p.k8sclient.CreateDataNodeDeployment(&count, baseImage, c.Spec.Zones[index], c.Spec.DataDiskSize, c.Spec.Resources, c.Spec.ImagePullSecrets, c.Metadata.Name, c.Spec.Instrumentation.StatsdHost)
+			p.k8sclient.CreateDataNodeDeployment(&count, baseImage, c.Spec.Zones[index], c.Spec.DataDiskSize, c.Spec.Resources, c.Spec.ImagePullSecrets, c.Metadata.Name, c.Spec.Instrumentation.StatsdHost, c.Spec.NetworkHost)
 		}
 	} else {
 		// No zones defined, rely on current provisioning logic which may break. Other strategy is to use emptyDir?
 		// NOTE: Issue with dynamic PV provisioning (https://github.com/kubernetes/kubernetes/issues/34583)
 		p.k8sclient.CreateStorageClass("standard", c.Spec.Storage.StorageClassProvisoner, c.Spec.Storage.StorageType, c.Metadata.Name)
-		p.k8sclient.CreateDataNodeDeployment(func() *int32 { i := int32(c.Spec.DataNodeReplicas); return &i }(), baseImage, "standard", c.Spec.DataDiskSize, c.Spec.Resources, c.Spec.ImagePullSecrets, c.Metadata.Name, c.Spec.Instrumentation.StatsdHost)
+		p.k8sclient.CreateDataNodeDeployment(func() *int32 { i := int32(c.Spec.DataNodeReplicas); return &i }(), baseImage, "standard", c.Spec.DataDiskSize, c.Spec.Resources, c.Spec.ImagePullSecrets, c.Metadata.Name, c.Spec.Instrumentation.StatsdHost, c.Spec.NetworkHost)
 	}
 
 	// Setup CronSchedule

--- a/pkg/spec/cluster.go
+++ b/pkg/spec/cluster.go
@@ -90,6 +90,9 @@ type ClusterSpec struct {
 	// Instrumentation defines metrics for the cluster
 	Instrumentation Instrumentation `json:"instrumentation"`
 	
+	// Specify how the container binds to network ports
+	NetworkHost string `json:"network-host"`
+	
 	//NodePort
 	NodePort int32 `json:"nodePort"`
 	

